mrjob/job.py

```diff
@@ -400,25 +406,26 @@ def increment_counter(self, group, counter, amount=1):
         if not isinstance(amount, (int, long)):
             raise TypeError('amount must be an integer, not %r' % (amount,))
 
+        # cast non-strings to strings (if people pass in exceptions, etc)
+        if not isinstance(group, basestring):
+            group = str(group)
+        if not isinstance(counter, basestring):
+            counter = str(counter)
+
```

Comment:
This stuff would be less weird (and have one less if-isinstance) if you invariantly made sure these things were str.